test/sys/test_socket.rs

```diff
@@ -2577,6 +2577,192 @@ fn test_recvmsg_rxq_ovfl() {
     assert_eq!(drop_counter, 1);
 }
 
+#[cfg(target_os = "linux")]
```

Comment:
I have added the socket option `IP_TOS` and `IPV6_TCLASS` for Android and FreeBSD in #2464, so hopefully these 2 tests should work on them as well, so:

```suggestion
#[cfg(any(linux_android, target_os = "freebsd"))]
```